src/main/java/com/yahoo/bullet/aggregations/TopK.java

```diff
@@ -53,8 +53,8 @@ public TopK(Aggregation aggregation, BulletConfig config) {
         int maxMapSize = config.getAs(BulletConfig.TOP_K_AGGREGATION_SKETCH_ENTRIES, Integer.class);
         Number threshold = getThreshold(attributes);
         int size = aggregation.getSize();
-        sketch = threshold != null ? new FrequentItemsSketch(errorType, maxMapSize, threshold.longValue(), size) :
-                                     new FrequentItemsSketch(errorType, maxMapSize, size);
+        sketch = threshold != null ? new FrequentItemsSketch(errorType, maxMapSize, threshold.longValue(), size, config.getBulletRecordProvider()) :
+                                     new FrequentItemsSketch(errorType, maxMapSize, size, config.getBulletRecordProvider());
```

Comment:
You can store a reference to the record provider above before the ternary since that seems to the be convention here